bigflow/dataflow.py

```diff
@@ -1,69 +1,204 @@
 import typing
 import logging
+import uuid
 
 from apache_beam import Pipeline
-from apache_beam.options.pipeline_options import PipelineOptions, GoogleCloudOptions
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.runners.runner import PipelineResult, PipelineState
 
-from bigflow.commons import public
+from typing import Dict, List, Union, Optional, Tuple
+
+try:
+    from typing import TypedDict, Literal
+except ImportError:
+    from typing_extensions import TypedDict, Literal
+
+from bigflow.commons import public, build_docker_image_tag
 from bigflow.workflow import Job, JobContext
-from bigflow.workflow import DEFAULT_EXECUTION_TIMEOUT_IN_SECONDS, DEFAULT_PIPELINE_LEVEL_EXECUTION_TIMEOUT_SHIFT_IN_SECONDS
+
+import bigflow.build.reflect
+
 
 logger = logging.getLogger(__file__)
 
 
+class PipelineOptionsDict(TypedDict, total=False):
```

Comment:
I think that you can also add mostly pre-configured pipeline options class for our internal users:
```
runner = 'DataflowRunner'
experiments = ['shuffle_mode=service']
machine_type = 'n2-standard-4'
max_num_workers = 200
disk_size_gb = 100
autoscaling_algorithm = 'THROUGHPUT_BASED'
network = 'default'
use_public_ips = False
region = 'europe-west1'
staging_location and temp_location = it would be nice to automatically create these in the future
setup_file = str(reflect.materialize_setuppy())
job_name = somehow infered from the BeamJob?
```

Reply:
Options 'setup_file' and 'job_name' are already provided in this PR (via method BeamJob.set_default_pipline_options).
I'll definitely add 'runner'...  Not sure about rest of parameters.